example/offboard_velocity/offboard_velocity.cpp

```diff
@@ -0,0 +1,266 @@
+/**
+* @file offboard_velocity.cpp
+* @brief Example that demonstrates offboard velocity control in local NED and body coordinates
+* @author Author: Julian Oes <julian@oes.ch>, Shakthi Prashanth <shakthi.prashanth.m@intel.com>
+* @date 2017-10-17
+*/
+
+#include <iostream>
+#include <cmath>
+#include <thread>
+#include <chrono>
+#include <dronecore/dronecore.h>
+
+using namespace dronecore;
+using std::this_thread::sleep_for;
+using std::chrono::milliseconds;
+using std::chrono::seconds;
+
+#define ERROR_CONSOLE_TEXT "\033[31m" //Turn text on console red
+#define TELEMETRY_CONSOLE_TEXT "\033[34m" //Turn text on console blue
+#define NORMAL_CONSOLE_TEXT "\033[0m"  //Restore normal console colour
+
+// Handles Action's result
+inline void action_error_exit(Action::Result result, const std::string &message)
+{
+    if (result != Action::Result::SUCCESS) {
+        std::cerr << ERROR_CONSOLE_TEXT << message << Action::result_str(
+                      result) << NORMAL_CONSOLE_TEXT << std::endl;
+        exit(EXIT_FAILURE);
+    }
+}
+
+// Handles Offboard's result
+inline void offboard_error_exit(Offboard::Result result, const std::string &message)
+{
+    if (result != Offboard::Result::SUCCESS) {
+        std::cerr << ERROR_CONSOLE_TEXT << message << Offboard::result_str(
+                      result) << NORMAL_CONSOLE_TEXT << std::endl;
+        exit(EXIT_FAILURE);
+    }
+}
+
+// Handles connection result
+inline void connection_error_exit(DroneCore::ConnectionResult result, const std::string &message)
+{
+    if (result != DroneCore::ConnectionResult::SUCCESS) {
+        std::cerr << ERROR_CONSOLE_TEXT << message
+                  << DroneCore::connection_result_str(result)
+                  << NORMAL_CONSOLE_TEXT << std::endl;
+        exit(EXIT_FAILURE);
+    }
+}
+
+// Logs during Offboard control
+inline void offboard_log(const std::string &offb_mode, const std::string msg)
+{
+    std::cout << "[" << offb_mode << "]" << msg << std::endl;
+}
+
+/**
+* @name connected device
+* @brief Does Offboard control using NED co-ordinates
+* returns true if everything went well in Offboard control, exits with a log otherwise.
+**/
+bool offb_ctrl_ned(Device &device)
+{
+    const std::string offb_mode = "NED";
+    // Send it once before starting offboard, otherwise it will be rejected.
+    device.offboard().set_velocity_ned({0.0f, 0.0f, 0.0f, 0.0f});
+    offboard_log(offb_mode, " Sent Null velocity command once before starting OFFBOARD");
+
+    Offboard::Result offboard_result = device.offboard().start();
+    offboard_error_exit(offboard_result, "Offboard start failed");
+    offboard_log(offb_mode, " OFFBOARD started");
+    sleep_for(seconds(1));
+
+    // Let yaw settle.
+    offboard_log(offb_mode, " Let yaw settle...");
+    for (unsigned i = 0; i < 100; ++i) {
+        device.offboard().set_velocity_ned({0.0f, 0.0f, 0.0f, 90.0f});
+        sleep_for(milliseconds(10));
```

Comment:
@julianoes Are we expecting the API client to maintain the resend rate? Not saying we shouldn't, just that we should think about whether there is any way to make this easier for developers. 
I don't think it matters too much for velocity control, but definitely for position control it would be much better if the developer could send a message "move to position" and not have to control resends.

Perhaps I am overthinking this.

Reply:
I agree that we might want to handle that internally.